### ISSUE TITLE
AllExpectations silences single skipped test

### ISSUE THREAD

**Issue by izcet** [etorreborre/specs2#783]
Take the following code example, `ExampleSpec.scala`
```
package a.b.c

import org.specs2.execute.Result
import org.specs2.specification.AllExpectationse
import org.specs2.specification.core.SpecStructure

class ExampleSpec extends Specification with AllExpectations {

  def is: SpecStructure = s2"""
    (2 + 2) = 4       $successExample
    (2 + 2) = fish    $skippedExample
    """

  def successExample(): Result = {
    2 + 2 must_==(4)
  }

  def skippedExample(): Result = {
    skipped("Todo: Implement numeric type for fish")
  }

}
```
If you compile and run this, you will get
```
[info] ExampleSpec
[info]     + (2 + 2) = 4
[info]     + (2 + 2) = fish
[info] Total for specification ExampleSpec
[info] Finished in 34 ms
[info] 2 examples, 1 expectation, 0 failure, 0 error
```
The sort of behavior that would be expected (or at least that I would expect) can be achieved if you remove the trait `with AllExpectations`
```
[info] ExampleSpec
[info]     + (2 + 2) = 4
[info]     o (2 + 2) = fish
[info] Todo: Implement numeric type for fish
[info] Total for specification ExampleSpec
[info] Finished in 8 ms
[info] 2 examples, 0 failure, 0 error, 1 skipped
```

I wasn't able to single out the source of the bug from some casual probing.

**Comment by etorreborre** [comment]
That's a good catch, I have a fix but I need to polish it. The source code for this feature is indeed quite complex, thanks to side-effects, which are the only way to do this magic of collecting all the expectations.

**Comment by etorreborre** [comment]
I pushed a fix to `master`, do you need it to be published soon?

**Comment by izcet** [comment]
Thanks @etorreborre! It's not urgent, but what's the time estimate for when it would get published?

**Comment by etorreborre** [comment]
I published `4.8.1`: https://repo1.maven.org/maven2/org/specs2/specs2-core_2.13/4.8.1/
